src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -378,23 +415,17 @@ internal ProjectId GetOrCreateProjectIdForPath(string projectPath, string projec
         }
 
         /// <summary>
-        /// Notifies the workspace host about the given action.
+        /// Notifies the <see cref="Workspace"/> of the change with the appropriate threading handling.
         /// </summary>
         /// <remarks>This method must be called on the foreground thread.</remarks>
-        internal void NotifyWorkspaceHosts(Action<IVisualStudioWorkspaceHost> action)
+        internal void NotifyWorkspace(Action<Workspace> action)
         {
             AssertIsForeground();
 
             // We do not want to allow message pumping/reentrancy when processing project system changes.
             using (Dispatcher.CurrentDispatcher.DisableProcessing())
```

Comment:
so this is to run action under dispatcher disable processing?

Reply:
Yep. Whether it's useful I don't actually know.